sa/sa.go

```diff
@@ -167,23 +167,27 @@ func (ssa *SQLStorageAuthority) GetRegistrationByKey(ctx context.Context, key *j
 }
 
 // GetAuthorization obtains an Authorization by ID
-func (ssa *SQLStorageAuthority) GetAuthorization(ctx context.Context, id string) (authz core.Authorization, err error) {
+func (ssa *SQLStorageAuthority) GetAuthorization(ctx context.Context, id string) (core.Authorization, error) {
+	authz := core.Authorization{}
 	tx, err := ssa.dbMap.Begin()
 	if err != nil {
-		return
+		return authz, err
 	}
-
 	pa, err := selectPendingAuthz(tx, "WHERE id = ?", id)
 	if err != nil && err != sql.ErrNoRows {
-		err = Rollback(tx, err)
-		return
+		return authz, Rollback(tx, err)
 	}
 	if err == sql.ErrNoRows {
 		var fa authzModel
-		err = tx.SelectOne(&fa, fmt.Sprintf("SELECT %s FROM authz WHERE id = ?", authzFields), id)
-		if err != nil {
-			err = Rollback(tx, err)
-			return
+		err := tx.SelectOne(&fa, fmt.Sprintf("SELECT %s FROM authz WHERE id = ?", authzFields), id)
+		if err != nil && err != sql.ErrNoRows {
+			return authz, Rollback(tx, err)
+		} else if err == sql.ErrNoRows {
+			rollbackErr := tx.Rollback()
```

Comment:
Since we've not actually done anything at this point is there a specific reason to rollback vs. commit? I'm not entirely sure there are significant performance differences (since the transaction will contain nothing and presumably hold no locks at this point) so I think it's more of a semantic thing...

Reply:
Also can be condensed into a single line.

Reply:
No specific reason to `Rollback` vs `Commit`. It just needs to be finalized one way or another.